src/synth/fluid_voice.c

```diff
@@ -663,22 +676,24 @@ calculate_hold_decay_buffers(fluid_voice_t* voice, int gen_base,
 void
 fluid_voice_update_param(fluid_voice_t* voice, int gen)
 {
-  double q_dB;
-  fluid_real_t x;
-  fluid_real_t y;
-  unsigned int count, z;
   // Alternate attenuation scale used by EMU10K1 cards when setting the attenuation at the preset or instrument level within the SoundFont bank.
   static const float ALT_ATTENUATION_SCALE = 0.4f;
+  unsigned int count, z;
+  fluid_real_t q_dB;
+  fluid_real_t x = fluid_voice_gen_value(voice, gen);
```

Comment:
I understand why you want to have this outside of the switch, but it does mean that on some generators this the gen value is calculated twice (on coarse/fine generators). Probably doesn't have a huge performance impact... but still, is it really worth it just to save a line of code in each `case`?

Reply:
> but it does mean that on some generators the gen value is calculated twice (on coarse/fine generators).

Yes. The generators in question should not be triggered too often and if they do, the required generator values are most likely to reside in L1 cache. So I thought that way it would be more readable as the majority of the generators use `x`. But if you insist I change it back.

Reply:
I don't (insist). It's probably fine.